clients/src/test/java/org/apache/kafka/common/network/NetworkTestUtils.java

```diff
@@ -87,6 +88,16 @@ public static void checkClientConnection(Selector selector, String node, int min
         }
     }
 
+    public static void waitForChannelConnected(Selector selector, String node) throws IOException {
+        int secondsLeft = 30;
+        while (selector.channel(node) != null
+                && !selector.channel(node).isConnected() && secondsLeft-- > 0) {
+            selector.poll(1000L);
+        }
+        assertNotNull(selector.channel(node));
+        assertTrue(selector.channel(node).isConnected());
```

Comment:
I'll add error message in the assertion, to indicate what we expected, ex:
`assertTrue(selector.channel(node).isConnected(), "channel " + node + " is not connected after 30 secs");`

Same to below `waitForChannelReady` method. (I know it's not your change, but let's improve it together)

Reply:
Agreed, thanks.